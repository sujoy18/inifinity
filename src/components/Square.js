//@flow
import React, {Component, type Node} from "react";

type Props = {
  black: bool,
  children: Node,
  onHandleClick: Function,
  position: Object
}
export default class Square extends Component < Props > {

  render() {
    const {black, position, children, onHandleClick} = this.props;
    const fill = black
      ? "black"
      : "white";
    const stroke = black
      ? "white"
      : "black";

    return (
      <div
        style={{
        backgroundColor: fill,
        color: stroke,
        width: "100%",
        height: "100%"
      }}
        onClick={() => {
        onHandleClick(position)
      }}>
        {children}
      </div>
    );
  }
}
