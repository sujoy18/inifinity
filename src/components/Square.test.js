import React from 'react';
import Square from './Square';

describe('Squre test', () => {

  it('should render correctly', () => {
    const wrapper = shallow(<Square black={true}/>);
    expect(wrapper).toMatchSnapshot();
    const div = wrapper.find('div');
    expect(div.prop('style')).toHaveProperty('backgroundColor', 'black');

  })

  it('should render div with white background', () => {
    const wrapper = shallow(<Square black={false}/>);
    expect(wrapper).toMatchSnapshot();
    const div = wrapper.find('div');
    expect(div.prop('style')).toHaveProperty('backgroundColor', 'white');

  })
})
