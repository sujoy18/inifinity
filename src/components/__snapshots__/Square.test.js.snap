// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Squre test should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Square
    black={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "style": Object {
        "backgroundColor": "black",
        "color": "white",
        "height": "100%",
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "style": Object {
          "backgroundColor": "black",
          "color": "white",
          "height": "100%",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Squre test should render div with white background 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Square
    black={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "style": Object {
        "backgroundColor": "white",
        "color": "black",
        "height": "100%",
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "style": Object {
          "backgroundColor": "white",
          "color": "black",
          "height": "100%",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
